### PR TITLE
[Merged by Bors] - feat: add a file tracking results from the 1000+ theorems project

### PR DESCRIPTION
**Pull request by grunweg**
Add a new file `1000.yaml`, tracking the formalisation status of theorems
from the [1000+ theorems project](https://1000-plus.github.io). Make `check-yaml` handle this file also.

Future PRs will
- expose this information on the webpage
- add infrastructure for synchronising this file with the upstream repository
- could add an attribute similar to the `stacks` attribute, and generate this file from the tagging
- add information on existing formalisation projects in Lean.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by jsm28** [comment]
The new file seems to be in a random order. Since this project doesn't have a fixed order (theorem 1, theorem 2, ...), I'd suggest sorting it either alphabetically by theorem name  (whether by Unicode collation or code-point order) or numerically by Wikidata code. (The 1000-plus data includes MSC codes, and use of those would be reasonable to provide more structure in the list on our website rather than having a purely alphabetical list there, but I'm not sure if sorting by MSC code would make so much sense for this source file.)

**Comment by grunweg** [comment]
Good catch! You were right (the old file depended on file system order in some sense); I've added sorting by wikidata ID now.

**Comment by fpvandoorn** [comment]
Some of the titles contain unicode-escapes, e.g. `\u2013`.

File: scripts/check-yaml.lean [hunk 1]
```diff
@@ -24,17 +24,17 @@ def readJsonFile (α) [FromJson α] (path : System.FilePath) : IO α := do
...
+      s!"Entries in `docs/{dir}.yaml` refer to N declaration(s) that don't exist. \
         Please correct the following:")
 
 def processDb (decls : ConstMap) : String × String → IO Bool
 | (file, msg) => do
   let lines := ← readJsonFile DBFile file
   let missing := lines.filter (fun l => !(decls.contains l.2))
   if 0 < missing.size then
-    IO.println msg
+    IO.println (msg.replace "N" s!"{missing.size}")
```
Comment (fpvandoorn) [inline scripts/check-yaml.lean:37]: Surely `String.replace` is not the right way to do this.
Instead, `databases` should just be `List String` (the first component), and then here we print the message with `{dir}` and `N` filled in.

**Comment by grunweg** [comment]
> Some of the titles contain unicode-escapes, e.g. `\u2013`.

That's added by the yaml generation script. I'll see if there's an option to output unicode instead.

File: scripts/check-yaml.lean (hunk 1 above)
Comment (grunweg) [reply scripts/check-yaml.lean:37]: Fixed and extracted to #19688.

**Comment by grunweg** [comment]
Updated and rebased now.

**Comment by grunweg** [comment]
For cross-reference: 1000-plus/1000-plus.github.io#6 and 1000-plus/1000-plus.github.io#7.

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**